### PR TITLE
fix: correct Button elevation

### PR DESCRIPTION
**Pull request by tasugi**
### Summary

It has a bug that Button elevation is still zero after mode is changed to `contained` from other modes.
This PR fixes this.

### Test plan

1. Render a Button with mode other than `contained`.
2. Change the mode to `contained` mode dynamically(e.g. on button press).
3. Check elevation

### CODE REVIEW

**Comment by brunohkbx** [comment]
related to #2736? cc @lukewalczak

**Comment by lukewalczak** [comment]
@tasugi could you please provide the snack to the issue which you're solving, where we can observe the bug?

**Comment by tasugi** [comment]
@lukewalczak Sure! I created a snack. https://snack.expo.io/@tasugi/reactnativepaper-pr2735

If you press the button, you can change the mode of it `contained`. I think you can see the button is not elevated.( Element inspector may help you to check it)

File: src/components/Button.tsx [hunk 1]
```diff
@@ -143,8 +143,9 @@ const Button = ({
   ...rest
 }: Props) => {
   const { current: elevation } = React.useRef<Animated.Value>(
-    new Animated.Value(mode === 'contained' ? 2 : 0)
+    new Animated.Value(0)
   );
+  elevation.setValue(mode === 'contained' ? 2 : 0);
```
Comment (lukewalczak) [inline src/components/Button.tsx:148]:
```suggestion
  React.useEffect(() => {
    elevation.setValue(mode === 'contained' ? 2 : 0);
  }, [mode, elevation]);
```

**Comment by tasugi** [comment]
@lukewalczak Thank you for suggestions!

File: src/components/Button.tsx [hunk 2]
```diff
@@ -143,8 +143,11 @@ const Button = ({
   ...rest
 }: Props) => {
   const { current: elevation } = React.useRef<Animated.Value>(
-    new Animated.Value(mode === 'contained' ? 2 : 0)
+    new Animated.Value(0)
```
Comment (lukewalczak) [inline src/components/Button.tsx:146]: I guess we have to revert that change to initially have elevation set to 2 for contained buttons

**Comment by tasugi** [comment]
Fixed an issue that caused the test to fail.

**Comment by lukewalczak** [comment]
Thanks @tasugi!

**Pull request merged**